test/upgrade/2.2.1-2.3.0.js

```diff
@@ -0,0 +1,1552 @@
+            beaconProxyAddress,
+            assistant.address,
+            ""
+          );
+
+          let bosonVoucher = await getContractAt("IBosonVoucher", expectedCloneAddress);
+
+          await expect(tx).to.emit(bosonVoucher, "ContractURIChanged");
+          await expect(tx).to.emit(bosonVoucher, "RoyaltyPercentageChanged");
+          bosonVoucher = await getContractAt("OwnableUpgradeable", expectedCloneAddress);
+          await expect(tx).to.emit(bosonVoucher, "OwnershipTransferred");
+        });
+      });
+
+      context("MetaTransactionHandler", async function () {
+        it("Function hashes from removedFunctionsHashes list should not be allowlisted", async function () {
+          for (const hash of removedFunctionHashes) {
+            // get function name from hash
+            const isFunctionAllowlisted = contractsAfter.metaTransactionsHandler.getFunction(
+              "isFunctionAllowlisted(bytes32)"
+            );
+            const isAllowed = await isFunctionAllowlisted.staticCall(hash);
+            expect(isAllowed).to.be.false;
+          }
+        });
+
+        it("Function hashes from from addedFunctionsHashes list should be allowlisted", async function () {
+          for (const hash of addedFunctionHashes) {
+            const isFunctionAllowlisted = await contractsAfter.metaTransactionsHandler.getFunction(
+              "isFunctionAllowlisted(bytes32)"
+            );
+
+            const isAllowed = await isFunctionAllowlisted.staticCall(hash);
+            expect(isAllowed).to.be.true;
+          }
+        });
+
+        it("State of metaTxPrivateContractState is not affected apart from isAllowlistedState mapping", async function () {
+          // make a shallow copy to not modify original protocolContractState as it's used on getGenericContext
+          const metaTxPrivateContractStateBefore = { ...protocolContractStateBefore.metaTxPrivateContractState };
+          const metaTxPrivateContractStateAfter = { ...protocolContractStateAfter.metaTxPrivateContractState };
+          const { isAllowlistedState: isAllowlistedStateBefore } = metaTxPrivateContractStateBefore;
+          removedFunctionHashes.forEach((hash) => {
+            delete isAllowlistedStateBefore[hash];
+          });
+
+          const { isAllowlistedState: isAllowlistedStateAfter } = metaTxPrivateContractStateAfter;
+          addedFunctionHashes.forEach((hash) => {
+            delete isAllowlistedStateAfter[hash];
+          });
+
+          delete metaTxPrivateContractStateBefore.isAllowlistedState;
+          delete metaTxPrivateContractStateAfter.isAllowlistedState;
+
+          expect(isAllowlistedStateAfter).to.deep.equal(isAllowlistedStateBefore);
+          expect(protocolContractStateAfter.metaTxPrivateContractState).to.deep.equal(
+            protocolContractStateBefore.metaTxPrivateContractState
+          );
+        });
+      });
+
+      context("BosonVoucher", async function () {
```

Comment:
In previous upgrades, we tested the boson voucher upgrade separately in `upgrade/client` folder.

If it's easier to keep it here (to not duplicate the setup), that's fine. But we should still include the generic voucher context from `test/upgrade/clients/01_generic.js`